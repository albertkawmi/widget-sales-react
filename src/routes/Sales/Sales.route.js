import React from 'react'
import Store from '../../components/Store'
import SalesTable from '../../components/SalesTable'

export default class Sales extends React.Component {
  componentWillMount() {
    const shouldFetch = !this.context.sales.length
    if (shouldFetch) this.context.fetchSales()
  }

  render() {
    return (
      <div>
        <h2 className="page-heading">All Sales</h2>
        <SalesTable sales={this.context.sales} />
      </div>
    )
  }
}

Sales.contextTypes = Store.childContextTypes
