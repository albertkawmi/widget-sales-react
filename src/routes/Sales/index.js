import Sales from './Sales.route'
export default Sales
