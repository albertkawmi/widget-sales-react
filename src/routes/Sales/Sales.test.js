/* eslint-env jest */
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import Sales from './Sales.route'

Enzyme.configure({ adapter: new Adapter() });

const salesRoute = Enzyme.shallow(
  <Router>
    <Sales />
  </Router>
)

describe('<Sales />', () => {
  it('renders without crashing', () => {
    expect(salesRoute).toBeTruthy()
  })

  it('has not been changed unintentionally', () => {
    expect(salesRoute).toMatchSnapshot()
  })
})
