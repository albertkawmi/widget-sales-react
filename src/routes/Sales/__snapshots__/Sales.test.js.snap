// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sales /> renders a sales table 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sales />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="page-heading"
>
          All Sales
</h2>,
        <SalesTable
          sales={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "All Sales",
          "className": "page-heading",
        },
        "ref": null,
        "rendered": "All Sales",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "sales": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="page-heading"
>
            All Sales
</h2>,
          <SalesTable
            sales={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "All Sales",
            "className": "page-heading",
          },
          "ref": null,
          "rendered": "All Sales",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "sales": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "fetchSales": [Function],
      "sales": Array [],
    },
  },
}
`;
